### PR TITLE
Added more documentation for auth

### PR DESCRIPTION
**Pull request by dbrasseur-aneo**
Fix https://github.com/aneoconsulting/ArmoniK/issues/616

### CODE REVIEW

File: Common/src/Auth/Authorization/AuthorizationPolicyProvider.cs [hunk 1]
```diff
@@ -73,13 +78,25 @@ public AuthorizationPolicyProvider(IOptionsMonitor<AuthenticatorOptions> options
...
+  /// </summary>
+  /// <returns></returns>
   public Task<AuthorizationPolicy> GetDefaultPolicyAsync()
     => Task.FromResult(new AuthorizationPolicyBuilder(Authenticator.SchemeName).RequireAuthenticatedUser()
                                                                                .Build());
 
+  /// <summary>
+  ///   Get a null authorization policy, falling back to other policies
+  /// </summary>
+  /// <returns></returns>
```
Comment (aneojgurhem) [inline Common/src/Auth/Authorization/AuthorizationPolicyProvider.cs:92]: idem

File: Common/src/Auth/Authorization/AuthorizationPolicyProvider.cs [hunk 2]
```diff
@@ -73,13 +78,25 @@ public AuthorizationPolicyProvider(IOptionsMonitor<AuthenticatorOptions> options
...
   public Task<AuthorizationPolicy?> GetFallbackPolicyAsync()
     => Task.FromResult<AuthorizationPolicy?>(null);
 
+  /// <summary>
+  ///   Get an authorization policy accepting accepting anything
+  /// </summary>
+  /// <returns></returns>
```
Comment (aneojgurhem) [inline Common/src/Auth/Authorization/AuthorizationPolicyProvider.cs:99]: idem

File: Common/src/Auth/Authorization/Permissions.cs [hunk 3]
```diff
@@ -54,6 +54,10 @@ public static class Permissions
   // Permissions list
   public static readonly ImmutableList<Permission> PermissionList = GetPermissionList();
 
+  /// <summary>
+  ///   Get the list of all permissions, based on the gRPC endpoints
+  /// </summary>
+  /// <returns></returns>
```
Comment (aneojgurhem) [inline Common/src/Auth/Authorization/Permissions.cs:60]: no returns

**Review by aneojgurhem — CHANGES_REQUESTED** [review]

File: Common/src/Auth/Authorization/AuthorizationPolicyProvider.cs [hunk 4]
```diff
@@ -73,13 +82,25 @@ public AuthorizationPolicyProvider(IOptionsMonitor<AuthenticatorOptions> options
...
 
+  /// <summary>
+  ///   Get a null authorization policy, falling back to other policies
+  /// </summary>
+  /// <returns>Null policy (fallback)</returns>
   public Task<AuthorizationPolicy?> GetFallbackPolicyAsync()
     => Task.FromResult<AuthorizationPolicy?>(null);
 
+  /// <summary>
+  ///   Get an authorization policy accepting accepting anything
```
Comment (aneojgurhem) [inline Common/src/Auth/Authorization/AuthorizationPolicyProvider.cs:101]: accepting x2

**Review by aneojgurhem — CHANGES_REQUESTED** [review]

File: Common/src/Auth/Authorization/AuthorizationPolicyProvider.cs (hunk 4 above)
Comment (dbrasseur-aneo) [reply Common/src/Auth/Authorization/AuthorizationPolicyProvider.cs:101]:
```suggestion
  ///   Get an authorization policy accepting anything
```

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**